### PR TITLE
Math Magicians: refactor with hooks

### PR DESCRIPTION
**Pull request by Iptysam**
- Refactor the Calculator component (components/Calculator.js) from a class-based component to a functional component with hooks.
- After the refactor is done, test the app in the browser. It should keep the same functionality

### CODE REVIEW

File: src/component/Calculator.js [hunk 1]
```diff
@@ -1,67 +1,62 @@
...
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-  }
+function Calculator() {
+  const [state, setState] = useState({
```
Comment (HAMMAS-SALEEM) [inline src/component/Calculator.js]: - [ ] Kindly convert your function into ES6 version as shown below :arrow_down: to follow the JavaScript best practices :+1:

**Your Function :red_circle:**
```
function Calculator() {
  ###
}

```
**ES6 Function :green_circle:**
```
const Calculator = () => {
  ###
}
```

File: src/component/Calculator.js [hunk 2]
```diff
@@ -1,67 +1,62 @@
...
-            <button type="button" onClick={this.handleClick}>+/-</button>
-            <button type="button" onClick={this.handleClick}>%</button>
-            <button type="button" onClick={this.handleClick} className="color">÷</button>
-          </div>
-          <div className="2nd row">
-            <button type="button" onClick={this.handleClick}>7</button>
-            <button type="button" onClick={this.handleClick}>8</button>
-            <button type="button" onClick={this.handleClick}>9</button>
-            <button type="button" onClick={this.handleClick} className="color">×</button>
-          </div>
```
Comment (HAMMAS-SALEEM) [inline src/component/Calculator.js:42]: - [ ] Your multiplication operation is not working. Kindly fix this to make your calculator fully usable :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @Iptysam :wave:

## Highlights :1st_place_medal:

:heavy_check_mark: Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
:heavy_check_mark: Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @Iptysam  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**